File: Models/BiomaSurfaceSWATSoilSWATC/src/cs/BiomaSurfaceSWATSoilSWATC/SurfaceSWATSoilSWATCState.cs
using System;
using System.Collections.Generic;
public class SurfaceSWATSoilSWATCState 
{
    private double[] _SoilTemperatureByLayers;
    
        public SurfaceSWATSoilSWATCState() { }
    
    
    public SurfaceSWATSoilSWATCState(SurfaceSWATSoilSWATCState toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    SoilTemperatureByLayers = new double[toCopy.SoilTemperatureByLayers.Length];
            for (int i = 0; i < toCopy.SoilTemperatureByLayers.Length; i++)
            { SoilTemperatureByLayers[i] = toCopy.SoilTemperatureByLayers[i]; }
    
    }
    }
    public double[] SoilTemperatureByLayers
        {
            get { return this._SoilTemperatureByLayers; }
            set { this._SoilTemperatureByLayers= value; } 
        }
}

File: Models/BiomaSurfaceSWATSoilSWATC/src/cpp/BiomaSurfaceSWATSoilSWATC/SurfaceSWATSoilSWATCState.cpp
#include "SurfaceSWATSoilSWATCState.h"
using namespace BiomaSurfaceSWATSoilSWATC;

SurfaceSWATSoilSWATCState::SurfaceSWATSoilSWATCState() {}

std::vector<double> & SurfaceSWATSoilSWATCState::getSoilTemperatureByLayers() { return this->SoilTemperatureByLayers; }

void SurfaceSWATSoilSWATCState::setSoilTemperatureByLayers(std::vector<double> const &_SoilTemperatureByLayers){
    this->SoilTemperatureByLayers = _SoilTemperatureByLayers;
}